### PR TITLE
docs: add mo-operator RFC

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [x] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

This RFC describes an overview of how will the operator work and be used.

Make sure you view this PR in preview mode to check the rendered graph https://github.com/matrixorigin/matrixone-operator/blob/0246d0d2bf496f0c79745e78aa4542caa26187d7/docs/rfc/2022-07-05-mo-cluster.md

### CODE REVIEW

**Review by wanglei4687 — APPROVED** [review]
:-) lgtm

File: docs/rfc/2022-07-05-mo-cluster.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    F -->|Create| D
+    F -->|Create| E
+    C -->|Watched| G
+    D -->|Watched| H
+    E -->|Watched| I
+```
+
+Physically, all the CRs are served by the kubernetes apiserver and all the controllers are running in goroutines of the operator process concurrently.
+
+Is is worth noting that `MatrixOneCluster` object is not necessary to create a mo-cluster, one can create `LogSet`, `CNSet` and `DNSet` separately to form a fully functional cluster. Such decoupling enables the following capabilities without losing the simplicity for end user:
```
Comment (sukki37) [inline docs/rfc/2022-07-05-mo-cluster.md:85]: It it worth nothing?

Comment (aylei) [reply docs/rfc/2022-07-05-mo-cluster.md:85]: It is not a typo, I mean note (noting)

**Pull request merged**